mailpoet/lib/Form/DisplayFormInWPContent.php

```diff
@@ -101,6 +103,16 @@ public function display($content = null) {
     return $result;
   }
 
+  public function contentDisplay($content = null) {
+    $this->inWooProductLoop = false;
+    return $this->display($content);
+  }
+
+  public function wooProductListDisplay($content = null) {
+    $this->inWooProductLoop = true;
+    return $this->display($content);
```

Comment:
We could change the display method from `public` to `private` to avoid calling it without setting the new property `inWooProductLoop`. But I am not sure how important is to have this property configured properly. What do you think?

Reply:
We can't change the method from `public` to `private`. It is being used here https://github.com/mailpoet/mailpoet/blob/40b0914d3c6864906de015f48ec8bf536a87abbb/mailpoet/lib/Config/Hooks.php#L211-L214

>  But I am not sure how important is to have this property configured properly.

We need the property in the check here https://github.com/mailpoet/mailpoet/blob/40b0914d3c6864906de015f48ec8bf536a87abbb/mailpoet/lib/Form/DisplayFormInWPContent.php#L128

Which help determine if we are within the `woocommerce_product_loop_end` filter or not.

Reply:
@triple0t I may have written it confusingly, but I thought changing `public function display($content = null)` to `private function display($content = null)`. Never mind, because it is a detail only.

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/4642/commits/e8109c66ce2cf711091e346643997006f41ed4c3